forms/src/main/java/org/devgateway/toolkit/forms/FormsSecurityConfig.java

```diff
@@ -99,4 +107,27 @@ protected void configure(final HttpSecurity http) throws Exception {
 				.httpStrictTransportSecurity().and().frameOptions().sameOrigin();
 
 	}
+
+	private SecurityExpressionHandler<FilterInvocation> webExpressionHandler() {
+		DefaultWebSecurityExpressionHandler defaultWebSecurityExpressionHandler = new DefaultWebSecurityExpressionHandler();
+		defaultWebSecurityExpressionHandler.setRoleHierarchy(roleHierarchy());
+		return defaultWebSecurityExpressionHandler;
+	}
+
+	/**
+	 * Defines role hierarchy.
+	 * Hierarchy is specified as a string. Space separates rules and > symbol has the meaning of 'includes'.
+	 * <p>Example: role1 > role2 > role3 role2 > role4</p>
+	 * <p>Here role1 includes role2 role3 and role4 (indirectly). And role2 includes role4.</p>
+	 *
+	 * <a href='http://docs.spring.io/spring-security/site/docs/4.1.2.RELEASE/reference/htmlsingle/#authz-hierarchical-roles'>Hierarchical Roles documentation</a>
+	 */
+	@Bean
+	RoleHierarchy roleHierarchy() {
+		RoleHierarchyImpl roleHierarchy = new RoleHierarchyImpl();
+		roleHierarchy.setHierarchy(
+				ROLE_ADMIN + " > " + ROLE_USER
```

Comment:
can we have the list of hierarchical roles externalized as a string maybe inside a properties file ? should help configuring it easier in forks